### PR TITLE
[SW-1387] Handle vectors in SparkDataFrameConverter more explicitly

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala [hunk 1]
```diff
@@ -341,20 +344,22 @@ object H2OSchemaUtils {
...
       } else {
         val head = flatDataFrame.head()
-        val sparseOrNot = vectorIndices.map { idx =>
-          head.getAs[org.apache.spark.mllib.linalg.Vector](idx).isInstanceOf[SparseVector]
-        }
-        vectorIndices.zip(sparseOrNot).toMap
+        vectorIndices.map { idx =>
+          head.get(idx) match {
+            case _: mllib.linalg.SparseVector => (idx, true)
+            case _: ml.linalg.SparseVector => (idx, true)
```
Comment (jakubhava) [inline core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala]: We had bug here -> we were collecting sparse info only from mllib.linalg.Vector

File: core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala [hunk 2]
```diff
@@ -109,26 +108,26 @@ object H2OSchemaUtils {
   }
 
   private def fillArray(
-      elementType: DataType,
-      flatSchemaIndexes: Map[String, Int],
-      buffer: ArrayBuffer[Any],
-      data: Any,
-      qualifiedName: String): Unit = {
+                         elementType: DataType,
```
Comment (mn-mikke) [inline core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala]: Do you prefer to use rather this formatting? The previous is based on coding conventions from Spark.  I am happy to use any type of formatting, but I would like to agree on which we will use in future.

File: core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala [hunk 3]
```diff
@@ -109,26 +108,26 @@ object H2OSchemaUtils {
...
+                         flatSchemaIndexes: Map[String, Int],
+                         buffer: ArrayBuffer[Any],
+                         data: Any,
+                         qualifiedName: String): Unit = {
     val seq = data.asInstanceOf[Seq[Any]]
     val subRow = Row.fromSeq(seq)
     val fillBufferPartiallyApplied = fillBuffer(flatSchemaIndexes, buffer) _
-    (0 until seq.size).foreach { idx =>
+    seq.indices.foreach { idx =>
```
Comment (mn-mikke) [inline core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala]: Nice!

File: core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala [hunk 4]
```diff
@@ -488,49 +483,40 @@ object H2OSchemaUtils {
...
-          case _: mllib.linalg.Vector =>
-            row.getAs[mllib.linalg.Vector](idx).size
-          case _: ml.linalg.Vector =>
-            row.getAs[ml.linalg.Vector](idx).size
-          case _ =>
-            throw new UnsupportedOperationException(s"User defined type is not supported: ${value.getClass}")
-        }
+      0
+    } else {
+      val dataType = row.schema.fields(idx).dataType
```
Comment (mn-mikke) [inline core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala]: Just a note: `schema` could be `null`

File: core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala (hunk 2 above)
Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala]: I always use the formatting by Idea after I click on `Reformat code` I don't have strong opinion on the formatting itself, but would like to be able to keep formatting using the Idea.

If there is any template idea can pick up and understand, I'm fine with that :)

File: core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala (hunk 3 above)
Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala]: Thanks IntelliJ :)

File: core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala (hunk 2 above)
Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala]: As we discussed on Slack, we are going to switch to formatting Spark is using, but in the following JIRA, thanks for bringing this up!

File: core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala (hunk 4 above)
Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala]: Good catch! Is it possible that row.schema is null even though whole dataFrame has schema defined?

**Pull request merged**